internal/storage/references/registry_reference.go

```diff
@@ -48,7 +59,7 @@ func ParseRegistryImageReferenceFromOutOfProcessData(input string) (RegistryImag
 		return RegistryImageReference{}, err
 	}
 	ref = reference.TagNameOnly(ref)
-	return RegistryImageReferenceFromRaw(ref), nil
+	return RegistryImageReferenceFromRaw(ref)
```

Comment:
The documentation of `ParseRegistryImageReferenceFromOutOfProcessData` should document that it removes the tag from tag@digest inputs.